test/context/onerror.js

```diff
@@ -88,6 +88,30 @@ describe('ctx.onerror(err)', function(){
     })
   })
 
+  it('should ignore error after headerSent', function(){
+    var app = koa();
+
+    app.on('error', function(err){
+      console.log(err);
+      err.message.should.equal('mock error');
+      err.headerSent.should.equal(true);
+      done();
+    });
+
+    app.use(function*() {
+      this.status = 200;
+      this.set('X-Foo', 'Bar');
+      this.res.flushHeaders();
+      yield Promise.reject(new Error('mock error'));
+      this.body = 'response';
+    });
+
+    request(app.listen())
+    .get('/')
+    .expect('X-Foo', 'Bar')
+    .expect(200, function() {});
```

Comment:
ignore the response, this maybe no response arrive because the response body is empty.